snapd-glib/snapd-change.c

```diff
@@ -144,6 +146,28 @@ snapd_change_get_tasks (SnapdChange *self)
     return self->tasks;
 }
 
+/**
+ * snapd_change_get_data:
+ * @self: a #SnapdChange.
+ *
+ * Get the data field for this change. It is a GHashMap with GList entries, each with.
```

Comment:
Looking at the snapd source, the form of data is `map[string]interface{}`, so we can't assume the contents are a list and the data depends on the change kind.

It seems like the right solution is to make a SnapdChangeData class, which can only access the field names and a SnapdAutoRefreshChangeData class (that inherits from SnapdChangeData) that parses the new fields.

Reply:
What about returning a GVariant/Qvariant with the data?

Reply:
I've found doing that tends to just push the problem to the app, and it's hard to use and easy to not handle unexpected cases. I think if we're sure that snapd will keep the data in a fixed format then adding methods to do the decoding for the app will be a more robust solution.

Reply:
Good point...

Reply:
I'll create a SnapdChangeData class that receives a Json object with the data and parses it internally, and with a different method for each key.

Reply:
Done. I didn't create a derived class from SnapdChangeData because the fields have the same format.

Reply:
Ok, after thinking a bit about it, I think that you are right: it's better to use base and child classes. Done.